lpvartypes.pas

```diff
@@ -4539,6 +4564,14 @@ function TLapeCompilerBase.hasDeclaration(ADecl: TLapeDeclaration; LocalOnly: Bo
   Result := hasDeclaration(ADecl, FStackInfo, LocalOnly, CheckWith);
 end;
 
+procedure TLapeCompilerBase.Hint(Msg: lpString; Args: array of const; ADocPos: TDocPos);
+begin
+  Assert(lcoHints in Options);
```

Comment:
With the way it works now (whether hints are on or off depends on the state during definition), should this check not be removed?